src/Castle.Facilities.Quartz.Tests/IntegrationTests/QuartzFacilityIntegrationTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Castle.Facilities.Startable;
+using Castle.MicroKernel.Registration;
+using Castle.Windsor;
+using NUnit.Framework;
+using Quartz;
+
+namespace Castle.Facilities.Quartz.Tests.IntegrationTests
+{
+    [TestFixture]
+    public class QuartzFacilityIntegrationTests
+    {
+        public const string SchedulerInstanceName = "IntegrationTestScheduler";
+        private IWindsorContainer _container;
+
+        public Dictionary<string, string> QuartzProperties => new Dictionary<string, string>
+        {
+            {"quartz.scheduler.instanceName", SchedulerInstanceName},
+            {"quartz.threadPool.type", "Quartz.Simpl.DefaultThreadPool, Quartz"},
+            {"quartz.threadPool.threadCount", "1"}
+        };
+
+        [SetUp]
+        public void OnSetup()
+        {
+            _container = CreateContainer();
+        }
+
+
+        private IWindsorContainer CreateContainer()
+        {
+            IWindsorContainer containerBuilder = new WindsorContainer()
+                .Register(Component.For<ITestJobListener>().ImplementedBy<TestJobListener>().LifestyleSingleton())
+                .Register(Component.For<ITestJob>().ImplementedBy<TestJob>().LifestyleTransient())
+                .AddFacility<StartableFacility>(q => q.DeferredStart());
+
+            return containerBuilder;
+        }
+
+        private void Sleep(int seconds)
+        {
+            // sleep ... and ... wait ...
+            var task = Task.Run(() => Thread.Sleep(seconds * 1000));
+            task.Wait();
+        }
+
+        [Test]
+        public void TestJobListener()
+        {
+            // Add Quartz
+            _container.AddFacility<QuartzFacility>(q =>
+                q.SetJobListeners(new JobListener(_container.Resolve<ITestJobListener>()))
+                    .SetProperties(QuartzProperties));
```

Comment:
I'd either do it as a properly named fluent interface, or do as most other facilities that just have normal C# properties, my preference would be normal properties.

e.g. WCF facility:

```csharp
container.AddFacility<WcfFacility>(f =>
{
    f.Clients.DefaultChannelPolicy = new MyPolicy();
});
```

Reply:
Facility is configured by using normal properties